modules/loader/src/main/java/com/opengamma/strata/loader/LoaderUtils.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2015 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.loader;
+
+import org.joda.beans.JodaBeanUtils;
+
+import com.opengamma.strata.basics.index.FxIndex;
+import com.opengamma.strata.basics.index.IborIndex;
+import com.opengamma.strata.basics.index.Index;
+import com.opengamma.strata.basics.index.OvernightIndex;
+import com.opengamma.strata.basics.index.PriceIndex;
+import com.opengamma.strata.collect.Messages;
+import com.opengamma.strata.math.impl.interpolation.DoubleQuadraticInterpolator1D;
+import com.opengamma.strata.math.impl.interpolation.ExponentialExtrapolator1D;
+import com.opengamma.strata.math.impl.interpolation.ExponentialInterpolator1D;
+import com.opengamma.strata.math.impl.interpolation.FlatExtrapolator1D;
+import com.opengamma.strata.math.impl.interpolation.LinearExtrapolator1D;
+import com.opengamma.strata.math.impl.interpolation.LinearInterpolator1D;
+import com.opengamma.strata.math.impl.interpolation.LogLinearInterpolator1D;
+import com.opengamma.strata.math.impl.interpolation.NaturalCubicSplineInterpolator1D;
+import com.opengamma.strata.math.impl.interpolation.QuadraticPolynomialLeftExtrapolator;
+import com.opengamma.strata.math.impl.interpolation.TimeSquareInterpolator1D;
+
+/**
+ * Contains utilities for loading market data from input files.
+ */
+public final class LoaderUtils {
+  
+  static {
+    JodaBeanUtils.registerMetaBean(DoubleQuadraticInterpolator1D.meta());
+    JodaBeanUtils.registerMetaBean(ExponentialInterpolator1D.meta());
+    JodaBeanUtils.registerMetaBean(LinearInterpolator1D.meta());
+    JodaBeanUtils.registerMetaBean(LogLinearInterpolator1D.meta());
+    JodaBeanUtils.registerMetaBean(NaturalCubicSplineInterpolator1D.meta());
+    JodaBeanUtils.registerMetaBean(TimeSquareInterpolator1D.meta());
+
+    JodaBeanUtils.registerMetaBean(ExponentialExtrapolator1D.meta());
+    JodaBeanUtils.registerMetaBean(FlatExtrapolator1D.meta());
+    JodaBeanUtils.registerMetaBean(LinearExtrapolator1D.meta());
+    JodaBeanUtils.registerMetaBean(QuadraticPolynomialLeftExtrapolator.meta());
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Attempts to locate a rate index by reference name.
+   * <p>
+   * This utility searches both {@link IborIndex} and {@link OvernightIndex}.
+   * 
+   * @param reference  the reference name
+   * @return the resolved rate index
+   */
+  public static Index findIndex(String reference) {
```

Comment:
I've just noticed the implementation can return other types - the Javadoc is wrong